**Linux** [Build 8277](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=72383&buildTypeId=WFPR) outcome was **FAILURE** using a merge of 9464f5f7f756e24dd703f10733ef2f4c29dd7a81
Summary: Tests failed: 4 (4 new), passed: 3209, ignored: 372 Build time: 00:54:26
### Failed tests
```
org.jboss.as.test.integration.messaging.jms.legacy.LegacyJMSTestCase.testSendAndReceiveFromLegacyQueue: java.lang.AssertionError: null
	at org.jboss.as.test.integration.messaging.jms.legacy.LegacyJMSTestCase.doSendAndReceive(LegacyJMSTestCase.java:168)
	at org.jboss.as.test.integration.messaging.jms.legacy.LegacyJMSTestCase.testSendAndReceiveFromLegacyQueue(LegacyJMSTestCase.java:127)
------- Stdout: -------
 [0m16:45:30,555 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 79) WFLYUT0022: Unregistered web context: /DependentMessagingDeploymentTestCase
 [0m [0m16:45:30,569 INFO  [org.jboss.weld.deployer] (MSC service thread 1-1) WFLYWELD0010: Stopping weld service for deployment DependentMessagingDeploymentTestCase.war
 [0m [0m16:45:30,572 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0028: Stopped deployment DependentMessagingDeploymentTestCase.war (runtime-name: DependentMessagingDeploymentTestCase.war) in 18ms
 [0m [0m16:45:30,580 INFO  [org.jboss.as.repository] (management-handler-thread - 4) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/basic/target/jbossas/standalone/data/content/56/d7b7ecaee9c92e205d5e4e6fff8008406114ae/content
 [0m [0m16:45:30,580 INFO  [org.jboss.as.server] (management-handler-thread - 4) WFLYSRV0009: Undeployed "DependentMessagingDeploymentTestCase.war" (runtime-name: "DependentMessagingDeploymentTestCase.war")
 [0m [0m16:45:30,580 INFO  [org.jboss.as.controller] (management-handler-thread - 4) WFLYCTL0183: Service status report
WFLYCTL0185:    Newly corrected services:
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".WeldBootstrapService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".WeldStartService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.VIEW."org.jboss.as.test.integration.messaging.jms.deployment.QueueMDB".MESSAGE_ENDPOINT (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.WeldInterceptorBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.QueueMDB.ejb.non-functional-timerservice (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.VIEW."org.jboss.as.test.integration.messaging.jms.deployment.TopicMDB".MESSAGE_ENDPOINT (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.WeldInterceptorBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component.TopicMDB.ejb.non-functional-timerservice (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."com.sun.faces.config.ConfigureListener".CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."com.sun.faces.config.ConfigureListener".JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."com.sun.faces.config.ConfigureListener".START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."com.sun.faces.config.ConfigureListener".WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.faces.webapp.FacetTag".CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.faces.webapp.FacetTag".JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.faces.webapp.FacetTag".START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.faces.webapp.FacetTag".WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.PermittedTaglibsTLV".CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.PermittedTaglibsTLV".JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.PermittedTaglibsTLV".START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.PermittedTaglibsTLV".WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.ScriptFreeTLV".CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.ScriptFreeTLV".JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.ScriptFreeTLV".START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."javax.servlet.jsp.jstl.tlv.ScriptFreeTLV".WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.as.test.integration.messaging.jms.deployment.MessagingServlet".CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.as.test.integration.messaging.jms.deployment.MessagingServlet".JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.as.test.integration.messaging.jms.deployment.MessagingServlet".START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.as.test.integration.messaging.jms.deployment.MessagingServlet".WeldInstantiator (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.weld.servlet.WeldInitialListener".CREATE (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.weld.servlet.WeldInitialListener".JndiBindingsService (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.weld.servlet.WeldInitialListener".START (no longer required)
      service jboss.deployment.unit."DependentMessagingDeploymentTestCase.war".component."org.jboss.weld.servlet.WeldInitialListener".WeldInstantiator (no longer required)


org.jboss.as.test.integration.messaging.jms.legacy.LegacyJMSTestCase.testSendAndReceiveFromLegacyTopic: java.lang.AssertionError: null
	at org.jboss.as.test.integration.messaging.jms.legacy.LegacyJMSTestCase.doSendAndReceive(LegacyJMSTestCase.java:168)
	at org.jboss.as.test.integration.messaging.jms.legacy.LegacyJMSTestCase.testSendAndReceiveFromLegacyTopic(LegacyJMSTestCase.java:132)
------- Stdout: -------
 [0m16:45:34,061 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 77) AMQ221003: trying to deploy queue jms.queue.LegacyJMSTestCase-Queue
 [0m16:45:34,066 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "add",
    "address" => [
        ("subsystem" => "messaging-activemq"),
        ("server" => "default"),
        ("jms-queue" => "LegacyJMSTestCase-Queue")
    ],
    "legacy-entries" => ["java:jboss/exported/legacy/jms/LegacyJMSTestCase-Queue"],
    "entries" => ["java:jboss/exported/jms/LegacyJMSTestCase-Queue"]
}
 [0m16:45:34,068 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 74) AMQ221003: trying to deploy queue jms.topic.LegacyJMSTestCase-Topic
 [0m16:45:34,073 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "add",
    "address" => [
        ("subsystem" => "messaging-activemq"),
        ("server" => "default"),
        ("jms-topic" => "LegacyJMSTestCase-Topic")
    ],
    "legacy-entries" => ["java:jboss/exported/legacy/jms/LegacyJMSTestCase-Topic"],
    "entries" => ["java:jboss/exported/jms/LegacyJMSTestCase-Topic"]
}
16:45:37,197 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "remove",
    "address" => [
        ("subsystem" => "messaging-activemq"),
        ("server" => "default"),
        ("jms-queue" => "LegacyJMSTestCase-Queue")
    ]
}
16:45:37,203 INFO  [org.jboss.as.test.integration.common.jms.JMSOperationsProvider] (main) Operation successful for update = {
    "operation" => "remove",
    "address" => [
        ("subsystem" => "messaging-activemq"),
        ("server" => "default"),
        ("jms-topic" => "LegacyJMSTestCase-Topic")
    ]
}


org.jboss.as.test.integration.domain.management.cli.JmsTestCase.testAddRemoveJmsQueue: java.lang.AssertionError: Failed to execute line 'jms-queue remove --profile=default --queue-address=testJmsQueue': org.jboss.as.cli.CommandLineException: {"WFLYDC0074: Operation failed or was rolled back on all servers. Server failures:" => {"server-group" => {"main-server-group" => {"host" => {"master" => {"main-one" => "WFLYCTL0158: Operation handler failed: java.lang.NullPointerException"},"slave" => {"main-three" => "WFLYCTL0158: Operation handler failed: java.lang.NullPointerException"}}}}}}
	at org.jboss.as.test.integration.management.util.CLIWrapper.sendLine(CLIWrapper.java:165)
	at org.jboss.as.test.integration.management.util.CLIWrapper.sendLine(CLIWrapper.java:187)
	at org.jboss.as.test.integration.domain.management.cli.JmsTestCase.testRemoveJmsQueue(JmsTestCase.java:94)
	at org.jboss.as.test.integration.domain.management.cli.JmsTestCase.testAddRemoveJmsQueue(JmsTestCase.java:64)
------- Stdout: -------
INFO  [org.xnio] XNIO version 3.3.2.Final
INFO  [org.xnio.nio] XNIO NIO Implementation Version 3.3.2.Final
INFO  [org.jboss.remoting] JBoss Remoting version 4.0.11.Final
 [0m17:19:12,174 INFO  [org.jboss.modules] (main) JBoss Modules version 1.4.4.Final
 [0m [0m17:19:12,363 INFO  [org.jboss.as.process.Host Controller.status] (main) WFLYPC0018: Starting process 'Host Controller'
 [0m[Host Controller]  [0m17:19:12,915 INFO  [org.jboss.modules] (main) JBoss Modules version 1.4.4.Final [0m
[Host Controller]  [0m [0m17:19:13,123 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final [0m
[Host Controller]  [0m [0m17:19:13,194 INFO  [org.jboss.as] (MSC service thread 1-4) WFLYSRV0049: WildFly Full 10.0.0.CR2-SNAPSHOT (WildFly Core 2.0.0.CR3) starting [0m
[Host Controller]  [0m [0m17:19:13,902 INFO  [org.jboss.as.host.controller] (Controller Boot Thread) WFLYHC0003: Creating http management service using network interface (management) port (9990) securePort (-1) [0m
[Host Controller]  [0m [0m17:19:13,903 INFO  [org.xnio] (MSC service thread 1-4) XNIO version 3.3.2.Final [0m
[Host Controller]  [0m [0m17:19:13,914 INFO  [org.xnio.nio] (MSC service thread 1-4) XNIO NIO Implementation Version 3.3.2.Final [0m
[Host Controller]  [0m [0m17:19:13,982 INFO  [org.jboss.remoting] (MSC service thread 1-4) JBoss Remoting version 4.0.11.Final [0m
[Host Controller]  [0m [0m17:19:14,022 INFO  [org.jboss.as.remoting] (MSC service thread 1-4) WFLYRMT0001: Listening on [::1]:9999 [0m
[Host Controller]  [0m [33m17:19:15,530 WARN  [org.jboss.as.jmx] (Controller Boot Thread) WFLYJMX0006: <jmx-connector/> is no longer supporting. <remoting-connector/> should be used instead to allow remote connections via JBoss Remoting. [0m
[Host Controller]  [33m [0m [33m17:19:15,640 WARN  [org.jboss.as.jmx] (Controller Boot Thread) WFLYJMX0006: <jmx-connector/> is no longer supporting. <remoting-connector/> should be used instead to allow remote connections via JBoss Remoting. [0m
[Host Controller]  [33m [0m [0m17:19:16,087 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'default-stack' in the resource at address '/profile=default/subsystem=jgroups' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m17:19:16,104 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'enabled' in the resource at address '/profile=default/subsystem=datasources/data-source=ExampleDS' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m17:19:16,121 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'default-stack' in the resource at address '/profile=other/subsystem=jgroups' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m17:19:16,134 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'enabled' in the resource at address '/profile=other/subsystem=datasources/data-source=ExampleDS' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m17:19:16,356 INFO  [org.jboss.as.host.controller] (Controller Boot Thread) WFLYHC0023: Starting server main-one [0m
 [0m17:19:16,413 INFO  [org.jboss.as.process.Server:main-one.status] (ProcessController-threads - 3) WFLYPC0018: Starting process 'Server:main-one'
 [0m[Server:main-one]  [0m17:19:16,788 INFO  [org.jboss.modules] (main) JBoss Modules version 1.4.4.Final [0m
[Server:main-one]  [0m [0m17:19:17,062 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final [0m
[Server:main-one]  [0m [0m17:19:17,154 INFO  [org.jboss.as] (MSC service thread 1-4) WFLYSRV0049: WildFly Full 10.0.0.CR2-SNAPSHOT (WildFly Core 2.0.0.CR3) starting [0m
[Server:main-one]  [0m [0m17:19:17,279 INFO  [org.xnio] (MSC service thread 1-2) XNIO version 3.3.2.Final [0m
[Server:main-one]  [0m [0m17:19:17,286 INFO  [org.xnio.nio] (MSC service thread 1-2) XNIO NIO Implementation Version 3.3.2.Final [0m
[Server:main-one]  [0m [0m17:19:17,310 INFO  [org.jboss.remoting] (MSC service thread 1-2) JBoss Remoting version 4.0.11.Final [0m
[Host Controller]  [0m [0m17:19:17,895 INFO  [org.jboss.as.host.controller] (Remoting "master:MANAGEMENT" task-16) WFLYHC0021: Server [Server:main-one] connected using connection [Channel ID 00f22306 (inbound) of Remoting connection 00253323 to /0:0:0:0:0:0:0:1:56570] [0m
[Host Controller]  [0m [0m17:19:17,945 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://[::1]:9990/management [0m
[Host Controller]  [0m [0m17:19:17,945 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://[::1]:9990 [0m
[Host Controller]  [0m [0m17:19:17,946 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 10.0.0.CR2-SNAPSHOT (WildFly Core 2.0.0.CR3) (Host Controller) started in 5382ms - Started 46 of 47 services (11 services are lazy, passive or on-demand) [0m
[Host Controller]  [0m [0m17:19:17,951 INFO  [org.jboss.as.host.controller] (server-registration-threads - 1) WFLYHC0020: Registering server main-one [0m
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] ServerIdentity{name=main-one, host=master, server-group=main-server-group} status is starting
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] ServerIdentity{name=main-one, host=master, server-group=main-server-group} status is starting
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] ServerIdentity{name=main-one, host=master, server-group=main-server-group} status is starting
[Server:main-one]  [0m [0m17:19:19,077 INFO  [org.jboss.as.controller.management-deprecated] (ServerService Thread Pool -- 24) WFLYCTL0028: Attribute 'enabled' in the resource at address '/subsystem=datasources/data-source=ExampleDS' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Server:main-one]  [0m [0m17:19:19,129 INFO  [org.jboss.as.controller.management-deprecated] (ServerService Thread Pool -- 11) WFLYCTL0028: Attribute 'default-stack' in the resource at address '/subsystem=jgroups' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Server:main-one]  [0m17:19:19,205 INFO  [org.wildfly.extension.mod_cluster] (ServerService Thread Pool -- 28) WFLYMODCLS0005: Mod_cluster uses default load balancer provider [0m
[Server:main-one] 17:19:19,222 INFO  [org.jboss.as.webservices] (ServerService Thread Pool -- 29) WFLYWS0002: Activating WebServices Extension
[Server:main-one] 17:19:19,318 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 34) WFLYJCA0004: Deploying JDBC-compliant driver class org.h2.Driver (version 1.3)
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] ServerIdentity{name=main-one, host=master, server-group=main-server-group} status is starting
[Server:main-one] 17:19:19,410 INFO  [org.jboss.as.connector] (MSC service thread 1-2) WFLYJCA0009: Starting JCA Subsystem (WildFly/IronJacamar 1.3.0.Final)
[Server:main-one] 17:19:19,412 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-3) WFLYJCA0018: Started Driver service with driver-name = h2
[Server:main-one] 17:19:19,503 INFO  [org.wildfly.extension.io] (ServerService Thread Pool -- 51) WFLYIO001: Worker 'default' has auto-configured to 4 core threads with 32 task threads based on your 2 available processors
[Server:main-one] 17:19:19,503 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 46) WFLYCLINF0001: Activating Infinispan subsystem.


org.jboss.as.test.integration.domain.management.cli.JmsTestCase.testAddRemoveJmsTopic: java.lang.AssertionError: Failed to execute line 'jms-topic remove --profile=default --topic-address=testJmsTopic': org.jboss.as.cli.CommandLineException: {"WFLYDC0074: Operation failed or was rolled back on all servers. Server failures:" => {"server-group" => {"main-server-group" => {"host" => {"master" => {"main-one" => "WFLYCTL0158: Operation handler failed: java.lang.NullPointerException"},"slave" => {"main-three" => "WFLYCTL0158: Operation handler failed: java.lang.NullPointerException"}}}}}}
	at org.jboss.as.test.integration.management.util.CLIWrapper.sendLine(CLIWrapper.java:165)
	at org.jboss.as.test.integration.management.util.CLIWrapper.sendLine(CLIWrapper.java:187)
	at org.jboss.as.test.integration.domain.management.cli.JmsTestCase.testRemoveJmsTopic(JmsTestCase.java:124)
	at org.jboss.as.test.integration.domain.management.cli.JmsTestCase.testAddRemoveJmsTopic(JmsTestCase.java:70)
------- Stdout: -------
INFO  [org.jboss.as.cli.CommandContext] [testTopic]
INFO  [org.jboss.as.cli.CommandContext] [testJmsTopic, testTopic]
[Server:main-three] 17:19:36,157 ERROR [org.jboss.as.controller.management-operation] (ServerService Thread Pool -- 16) WFLYCTL0013: Operation ("remove") failed - address: ([
[Server:main-three]     ("subsystem" => "messaging-activemq"),
[Server:main-three]     ("server" => "default"),
[Server:main-three]     ("jms-topic" => "testJmsTopic")
[Server:main-three] ]): java.lang.NullPointerException
[Server:main-three] 	at org.wildfly.extension.messaging.activemq.jms.JMSTopicRemove.performRuntime(JMSTopicRemove.java:59)
[Server:main-three] 	at org.jboss.as.controller.AbstractRemoveStepHandler$1.execute(AbstractRemoveStepHandler.java:80) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.AbstractOperationContext.executeStep(AbstractOperationContext.java:877) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:651) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:362) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1336) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.ModelControllerImpl.internalExecute(ModelControllerImpl.java:391) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.ModelControllerImpl.execute(ModelControllerImpl.java:217) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler.internalExecute(TransactionalProtocolOperationHandler.java:233) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler$ExecuteRequestHandler.doExecute(TransactionalProtocolOperationHandler.java:173) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler$ExecuteRequestHandler$1.run(TransactionalProtocolOperationHandler.java:136) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler$ExecuteRequestHandler$1.run(TransactionalProtocolOperationHandler.java:132) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at java.security.AccessController.doPrivileged(Native Method) [rt.jar:1.8.0_40]
[Server:main-three] 	at javax.security.auth.Subject.doAs(Subject.java:360) [rt.jar:1.8.0_40]
[Server:main-three] 	at org.jboss.as.controller.AccessAuditContext.doAs(AccessAuditContext.java:81) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler$ExecuteRequestHandler$2$1.run(TransactionalProtocolOperationHandler.java:152) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler$ExecuteRequestHandler$2$1.run(TransactionalProtocolOperationHandler.java:148) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at java.security.AccessController.doPrivileged(Native Method) [rt.jar:1.8.0_40]
[Server:main-three] 	at org.jboss.as.controller.remote.TransactionalProtocolOperationHandler$ExecuteRequestHandler$2.execute(TransactionalProtocolOperationHandler.java:148) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.protocol.mgmt.AbstractMessageHandler$2$1.doExecute(AbstractMessageHandler.java:298) [wildfly-protocol-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at org.jboss.as.protocol.mgmt.AbstractMessageHandler$AsyncTaskRunner.run(AbstractMessageHandler.java:518) [wildfly-protocol-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-three] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_40]
[Server:main-three] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_40]
[Server:main-three] 	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_40]
[Server:main-three] 	at org.jboss.threads.JBossThread.run(JBossThread.java:320) [jboss-threads-2.2.1.Final.jar:2.2.1.Final]
[Server:main-three] 
[Server:main-one] 17:19:36,186 ERROR [org.jboss.as.controller.management-operation] (ServerService Thread Pool -- 17) WFLYCTL0013: Operation ("remove") failed - address: ([
[Server:main-one]     ("subsystem" => "messaging-activemq"),
[Server:main-one]     ("server" => "default"),
[Server:main-one]     ("jms-topic" => "testJmsTopic")
[Server:main-one] ]): java.lang.NullPointerException
[Server:main-one] 	at org.wildfly.extension.messaging.activemq.jms.JMSTopicRemove.performRuntime(JMSTopicRemove.java:59)
[Server:main-one] 	at org.jboss.as.controller.AbstractRemoveStepHandler$1.execute(AbstractRemoveStepHandler.java:80) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-one] 	at org.jboss.as.controller.AbstractOperationContext.executeStep(AbstractOperationContext.java:877) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-one] 	at org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:651) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-one] 	at org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:362) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-one] 	at org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1336) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]
[Server:main-one] 	at org.jboss.as.controller.ModelControllerImpl.internalExecute(ModelControllerImpl.java:391) [wildfly-controller-2.0.0.CR3.jar:2.0.0.CR3]


```

--- a/legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java
+++ b/legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java
@@ -417,7 +417,11 @@ public class MigrateOperation implements OperationStepHandler {
                                 String name = address.getLastElement().getValue();
                                 ModelNode value = newAddOp.get(VALUE);
                                 ModelNode parentAddOp = newAddOperations.get(address.getParent());
-                                parentAddOp.get("params").add(new Property(name, value));
+                                if (name.equals("http-upgrade-endpoint") && address.getParent().getLastElement().getKey().equals("http-connector")) {
+                                    parentAddOp.get("endpoint").set(value);
+                                } else {
+                                    parentAddOp.get("params").add(new Property(name, value));
+                                }
                                 continue;
                             }
                             break;

--- a/legacy/messaging/src/test/resources/org/jboss/as/messaging/test/subsystem_migration.xml
+++ b/legacy/messaging/src/test/resources/org/jboss/as/messaging/test/subsystem_migration.xml
@@ -73,6 +73,7 @@
         <connectors>
             <http-connector name="http" socket-binding="http">
                 <param key="batch-delay" value="${batch.delay:50}"/>
+                <param key="http-upgrade-endpoint" value="http"/>
             </http-connector>
             <netty-connector name="netty" socket-binding="messaging" />
             <netty-connector name="netty-throughput" socket-binding="messaging-throughput">

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPConnectorDefinition.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPConnectorDefinition.java
@@ -32,6 +32,7 @@ import org.jboss.as.controller.AttributeMarshaller;
 import org.jboss.as.controller.SimpleAttributeDefinition;
 import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraintDefinition;
 import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
 
 /**
  * HTTP connector resource definition
@@ -54,9 +55,15 @@ public class HTTPConnectorDefinition extends AbstractTransportDefinition {
             .addAccessConstraint(SensitiveTargetAccessConstraintDefinition.SOCKET_BINDING_REF)
             .build();
 
+    // for remote acceptor, the socket-binding is required
+    public static final SimpleAttributeDefinition ENDPOINT = create("endpoint", ModelType.STRING)
+            .setAllowNull(false)
+            .setAllowExpression(false) // references another resource
+            .build();
+
     static final HTTPConnectorDefinition INSTANCE = new HTTPConnectorDefinition();
 
     public HTTPConnectorDefinition() {
-        super(false, CommonAttributes.HTTP_CONNECTOR, SOCKET_BINDING, PARAMS);
+        super(false, CommonAttributes.HTTP_CONNECTOR, SOCKET_BINDING, ENDPOINT, PARAMS);
     }
 }

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPUpgradeService.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/HTTPUpgradeService.java
@@ -117,7 +117,7 @@ public class HTTPUpgradeService implements Service<HTTPUpgradeService> {
                     public boolean handleUpgrade(HttpServerExchange exchange) throws IOException {
 
                         if (super.handleUpgrade(exchange)) {
-                            final String endpoint = exchange.getRequestHeaders().getFirst(HTTP_UPGRADE_ENDPOINT_PROP_NAME);
+                            final String endpoint = exchange.getRequestHeaders().getFirst(getHttpUpgradeEndpointKey());
                             if (endpoint == null) {
                                 return true;
                             } else {
@@ -169,6 +169,11 @@ public class HTTPUpgradeService implements Service<HTTPUpgradeService> {
         return SEC_ACTIVEMQ_REMOTING_ACCEPT;
     }
 
+    protected String getHttpUpgradeEndpointKey() {
+        return HTTP_UPGRADE_ENDPOINT_PROP_NAME;
+    }
+
+
     /**
      * Service to handle HTTP upgrade for legacy (HornetQ) clients.
      *
@@ -195,6 +200,11 @@ public class HTTPUpgradeService implements Service<HTTPUpgradeService> {
         @Override
         protected String getProtocol() {
             return HORNETQ_REMOTING;
+        }
+
+        @Override
+        protected String getHttpUpgradeEndpointKey() {
+            return "http-upgrade-endpoint";
         }
 
         @Override

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_1_0.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingSubsystemParser_1_0.java
@@ -294,6 +294,7 @@ public class MessagingSubsystemParser_1_0 implements XMLStreamConstants, XMLElem
                                         builder(HTTPConnectorDefinition.INSTANCE)
                                                 .addAttributes(
                                                         HTTPConnectorDefinition.SOCKET_BINDING,
+                                                        HTTPConnectorDefinition.ENDPOINT,
                                                         CommonAttributes.PARAMS))
                                 .addChild(
                                         builder(RemoteTransportDefinition.CONNECTOR_INSTANCE)

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/TransportConfigOperationHandlers.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/TransportConfigOperationHandlers.java
@@ -22,6 +22,7 @@
 
 package org.wildfly.extension.messaging.activemq;
 
+import static org.apache.activemq.artemis.core.remoting.impl.invm.TransportConstants.SERVER_ID_PROP_NAME;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.ACCEPTOR;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.CONNECTOR;
 import static org.wildfly.extension.messaging.activemq.CommonAttributes.FACTORY_CLASS;
@@ -48,12 +49,164 @@ import org.jboss.as.controller.OperationContext;
 import org.jboss.as.controller.OperationFailedException;
 import org.jboss.dmr.ModelNode;
 import org.jboss.dmr.Property;
+
 /**
  * Basic {@link TransportConfiguration} (Acceptor/Connector) related operations.
+ *
+ * Artemis changed the naming convention for naming its parameters and uses CamelCase names.
+ * WildFly convention is to use hyphen-separated names. The mapping is done when creating Artemis connector/acceptor
+ * configuration based on the WildFly management model.
  *
  * @author Emanuel Muckenhuber
  */
 public class TransportConfigOperationHandlers {
+
+    private static final Map<String, String> CONNECTORS_KEYS_MAP = new HashMap<>();
+    private static final Map<String, String> ACCEPTOR_KEYS_MAP = new HashMap<>();
+
+    private static final String BATCH_DELAY = "batch-delay";
+    private static final String HTTP_UPGRADE_ENABLED = "http-upgrade-enabled";
+    private static final String KEY_STORE_PASSWORD = "key-store-password";
+    private static final String KEY_STORE_PATH = "key-store-path";
+    private static final String KEY_STORE_PROVIDER = "key-store-provider";
+    private static final String TCP_RECEIVE_BUFFER_SIZE = "tcp-receive-buffer-size";
+    private static final String TCP_SEND_BUFFER_SIZE = "tcp-send-buffer-size";
+    private static final String TRUST_STORE_PASSWORD = "trust-store-password";
+    private static final String TRUST_STORE_PATH = "trust-store-path";
+    private static final String TRUST_STORE_PROVIDER = "trust-store-provider";
+    private static final String ENABLED_PROTOCOLS = "enabled-protocols";
+    private static final String ENABLED_CIPHER_SUITES = "enabled-cipher-suites";
+    private static final String HOST = "host";
+    private static final String PORT = "port";
+    public static final String SSL_ENABLED = "ssl-enabled";
+    public static final String USE_NIO = "use-nio";
+    public static final String TCP_NO_DELAY = "tcp-no-delay";
+
+    static {
+        CONNECTORS_KEYS_MAP.put(InVMTransportDefinition.SERVER_ID.getName(),
+                org.apache.activemq.artemis.core.remoting.impl.invm.TransportConstants.SERVER_ID_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(SSL_ENABLED,
+                TransportConstants.SSL_ENABLED_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("http-enabled",
+                TransportConstants.HTTP_ENABLED_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("http-client-idle-time",
+                TransportConstants.HTTP_CLIENT_IDLE_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("http-client-idle-scan-period",
+                TransportConstants.HTTP_CLIENT_IDLE_SCAN_PERIOD);
+        CONNECTORS_KEYS_MAP.put("http-requires-session-id",
+                TransportConstants.HTTP_REQUIRES_SESSION_ID);
+        CONNECTORS_KEYS_MAP.put(HTTP_UPGRADE_ENABLED,
+                TransportConstants.HTTP_UPGRADE_ENABLED_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("http-upgrade-endpoint",
+                TransportConstants.HTTP_UPGRADE_ENDPOINT_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("use-servlet",
+                TransportConstants.USE_SERVLET_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("servlet-path",
+                TransportConstants.SERVLET_PATH);
+        CONNECTORS_KEYS_MAP.put(USE_NIO,
+                TransportConstants.USE_NIO_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("use-nio-global-worker-pool",
+                TransportConstants.USE_NIO_GLOBAL_WORKER_POOL_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(HOST,
+                TransportConstants.HOST_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(PORT,
+                TransportConstants.PORT_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("local-address",
+                TransportConstants.LOCAL_ADDRESS_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put("local-port",
+                TransportConstants.LOCAL_PORT_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(KEY_STORE_PROVIDER,
+                TransportConstants.KEYSTORE_PROVIDER_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(KEY_STORE_PATH,
+                TransportConstants.KEYSTORE_PATH_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(KEY_STORE_PASSWORD,
+                TransportConstants.KEYSTORE_PASSWORD_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(TRUST_STORE_PROVIDER,
+                TransportConstants.TRUSTSTORE_PROVIDER_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(TRUST_STORE_PATH,
+                TransportConstants.TRUSTSTORE_PATH_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(TRUST_STORE_PASSWORD,
+                TransportConstants.TRUSTSTORE_PASSWORD_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(ENABLED_CIPHER_SUITES,
+                TransportConstants.ENABLED_CIPHER_SUITES_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(ENABLED_PROTOCOLS,
+                TransportConstants.ENABLED_PROTOCOLS_PROP_NAME);
+        CONNECTORS_KEYS_MAP.put(TCP_NO_DELAY,
+                TransportConstants.TCP_NODELAY_PROPNAME);
+        CONNECTORS_KEYS_MAP.put(TCP_SEND_BUFFER_SIZE,
+                TransportConstants.TCP_SENDBUFFER_SIZE_PROPNAME);
+        CONNECTORS_KEYS_MAP.put(TCP_RECEIVE_BUFFER_SIZE,
+                TransportConstants.TCP_RECEIVEBUFFER_SIZE_PROPNAME);
+        CONNECTORS_KEYS_MAP.put("nio-remoting-threads",
+                TransportConstants.NIO_REMOTING_THREADS_PROPNAME);
+        CONNECTORS_KEYS_MAP.put(BATCH_DELAY,
+                TransportConstants.BATCH_DELAY);
+        CONNECTORS_KEYS_MAP.put("connect-timeout-millis",
+                TransportConstants.NETTY_CONNECT_TIMEOUT);
+
+        ACCEPTOR_KEYS_MAP.put(InVMTransportDefinition.SERVER_ID.getName(),
+                org.apache.activemq.artemis.core.remoting.impl.invm.TransportConstants.SERVER_ID_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(BATCH_DELAY,
+                TransportConstants.BATCH_DELAY);
+        ACCEPTOR_KEYS_MAP.put("cluster-connection",
+                TransportConstants.CLUSTER_CONNECTION);
+        ACCEPTOR_KEYS_MAP.put("connection-ttl",
+                TransportConstants.CONNECTION_TTL);
+        ACCEPTOR_KEYS_MAP.put("connections-allowed",
+                TransportConstants.CONNECTIONS_ALLOWED);
+        ACCEPTOR_KEYS_MAP.put("direct-deliver",
+                TransportConstants.DIRECT_DELIVER);
+        ACCEPTOR_KEYS_MAP.put(ENABLED_CIPHER_SUITES,
+                TransportConstants.ENABLED_CIPHER_SUITES_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(ENABLED_PROTOCOLS,
+                TransportConstants.ENABLED_PROTOCOLS_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(HOST,
+                TransportConstants.HOST_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("http-response-time",
+                TransportConstants.HTTP_RESPONSE_TIME_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("http-server-scan-period",
+                TransportConstants.HTTP_SERVER_SCAN_PERIOD_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(HTTP_UPGRADE_ENABLED,
+                TransportConstants.HTTP_UPGRADE_ENABLED_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(KEY_STORE_PASSWORD,
+                TransportConstants.KEYSTORE_PASSWORD_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(KEY_STORE_PATH,
+                TransportConstants.KEYSTORE_PATH_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(KEY_STORE_PROVIDER,
+                TransportConstants.KEYSTORE_PROVIDER_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("needs-client-auth",
+                TransportConstants.NEED_CLIENT_AUTH_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("nio-remoting-threads",
+                TransportConstants.NIO_REMOTING_THREADS_PROPNAME);
+        ACCEPTOR_KEYS_MAP.put(PORT,
+                TransportConstants.PORT_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("protocols",
+                TransportConstants.PROTOCOLS_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(SSL_ENABLED,
+                TransportConstants.SSL_ENABLED_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("stomp-enable-message-id",
+                TransportConstants.STOMP_ENABLE_MESSAGE_ID);
+        ACCEPTOR_KEYS_MAP.put("stomp-min-large-message-size",
+                TransportConstants.STOMP_MIN_LARGE_MESSAGE_SIZE);
+        ACCEPTOR_KEYS_MAP.put("stomp-consumer-credits",
+                TransportConstants.STOMP_CONSUMERS_CREDIT);
+        ACCEPTOR_KEYS_MAP.put(TCP_NO_DELAY,
+                TransportConstants.TCP_NODELAY_PROPNAME);
+        ACCEPTOR_KEYS_MAP.put(TCP_RECEIVE_BUFFER_SIZE,
+                TransportConstants.TCP_RECEIVEBUFFER_SIZE_PROPNAME);
+        ACCEPTOR_KEYS_MAP.put(TCP_SEND_BUFFER_SIZE,
+                TransportConstants.TCP_SENDBUFFER_SIZE_PROPNAME);
+        ACCEPTOR_KEYS_MAP.put(TRUST_STORE_PASSWORD,
+                TransportConstants.TRUSTSTORE_PASSWORD_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(TRUST_STORE_PATH,
+                TransportConstants.TRUSTSTORE_PATH_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(TRUST_STORE_PROVIDER,
+                TransportConstants.TRUSTSTORE_PROVIDER_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put("use-invm",
+                TransportConstants.USE_INVM_PROP_NAME);
+        ACCEPTOR_KEYS_MAP.put(USE_NIO,
+                TransportConstants.USE_NIO_PROP_NAME);
+    }
 
     /**
      * Process the acceptor information.
@@ -70,7 +223,7 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(ACCEPTOR).asPropertyList()) {
                 final String acceptorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
+                final Map<String, Object> parameters = getParameters(context, config, ACCEPTOR_KEYS_MAP);
                 final String clazz = config.get(FACTORY_CLASS.getName()).asString();
                 acceptors.put(acceptorName, new TransportConfiguration(clazz, parameters, acceptorName));
             }
@@ -79,10 +232,11 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(REMOTE_ACCEPTOR).asPropertyList()) {
                 final String acceptorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
+                final Map<String, Object> parameters = getParameters(context, config, ACCEPTOR_KEYS_MAP);
                 final String binding = config.get(RemoteTransportDefinition.SOCKET_BINDING.getName()).asString();
-                parameters.put(RemoteTransportDefinition.SOCKET_BINDING.getName(), binding);
                 bindings.add(binding);
+                // uses the parameters to pass the socket binding name that will be read in ActiveMQServerService.start()
+                parameters.put(RemoteTransportDefinition.SOCKET_BINDING.getName(), binding);
                 acceptors.put(acceptorName, new TransportConfiguration(NettyAcceptorFactory.class.getName(), parameters, acceptorName));
             }
         }
@@ -90,8 +244,8 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(IN_VM_ACCEPTOR).asPropertyList()) {
                 final String acceptorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
-                parameters.put(InVMTransportDefinition.SERVER_ID.getName(), InVMTransportDefinition.SERVER_ID.resolveModelAttribute(context, config).asInt());
+                final Map<String, Object> parameters = getParameters(context, config, ACCEPTOR_KEYS_MAP);
+                parameters.put(SERVER_ID_PROP_NAME, InVMTransportDefinition.SERVER_ID.resolveModelAttribute(context, config).asInt());
                 acceptors.put(acceptorName, new TransportConfiguration(InVMAcceptorFactory.class.getName(), parameters, acceptorName));
             }
         }
@@ -99,7 +253,7 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(HTTP_ACCEPTOR).asPropertyList()) {
                 final String acceptorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
+                final Map<String, Object> parameters = getParameters(context, config, ACCEPTOR_KEYS_MAP);
                 parameters.put(TransportConstants.HTTP_UPGRADE_ENABLED_PROP_NAME, true);
                 acceptors.put(acceptorName, new TransportConfiguration(NettyAcceptorFactory.class.getName(), parameters, acceptorName));
             }
@@ -112,12 +266,19 @@ public class TransportConfigOperationHandlers {
      *
      * @param context the operation context
      * @param config the transport configuration
+     * @param mapping Mapping betwen WildFly parameters (keys) and Artemis constants (values)
      * @return the extracted parameters
      * @throws OperationFailedException if an expression can not be resolved
      */
-    public static Map<String, Object> getParameters(final OperationContext context, final ModelNode config) throws OperationFailedException {
+    public static Map<String, Object> getParameters(final OperationContext context, final ModelNode config, final Map<String, String> mapping) throws OperationFailedException {
         Map<String, String> fromModel = CommonAttributes.PARAMS.unwrap(context, config);
-        return new HashMap<String, Object>(fromModel);
+        Map<String, Object> parameters = new HashMap<>();
+        for (Map.Entry<String, String> entry : fromModel.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            parameters.put(mapping.getOrDefault(key, key), value);
+        }
+        return parameters;
     }
 
     /**
@@ -135,7 +296,7 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(CONNECTOR).asPropertyList()) {
                 final String connectorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
+                final Map<String, Object> parameters = getParameters(context, config, CONNECTORS_KEYS_MAP);
                 final String clazz = FACTORY_CLASS.resolveModelAttribute(context, config).asString();
                 connectors.put(connectorName, new TransportConfiguration(clazz, parameters, connectorName));
             }
@@ -144,10 +305,11 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(REMOTE_CONNECTOR).asPropertyList()) {
                 final String connectorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
+                final Map<String, Object> parameters = getParameters(context, config, CONNECTORS_KEYS_MAP);
                 final String binding = config.get(RemoteTransportDefinition.SOCKET_BINDING.getName()).asString();
-                parameters.put(RemoteTransportDefinition.SOCKET_BINDING.getName(), binding);
                 bindings.add(binding);
+                // uses the parameters to pass the socket binding name that will be read in ActiveMQServerService.start()
+                parameters.put(RemoteTransportDefinition.SOCKET_BINDING.getName(), binding);
                 connectors.put(connectorName, new TransportConfiguration(NettyConnectorFactory.class.getName(), parameters, connectorName));
             }
         }
@@ -155,8 +317,8 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(IN_VM_CONNECTOR).asPropertyList()) {
                 final String connectorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
-                parameters.put(InVMTransportDefinition.SERVER_ID.getName(), InVMTransportDefinition.SERVER_ID.resolveModelAttribute(context, config).asInt());
+                final Map<String, Object> parameters = getParameters(context, config, CONNECTORS_KEYS_MAP);
+                parameters.put(CONNECTORS_KEYS_MAP.get(InVMTransportDefinition.SERVER_ID.getName()), InVMTransportDefinition.SERVER_ID.resolveModelAttribute(context, config).asInt());
                 connectors.put(connectorName, new TransportConfiguration(InVMConnectorFactory.class.getName(), parameters, connectorName));
             }
         }
@@ -164,12 +326,14 @@ public class TransportConfigOperationHandlers {
             for (final Property property : params.get(HTTP_CONNECTOR).asPropertyList()) {
                 final String connectorName = property.getName();
                 final ModelNode config = property.getValue();
-                final Map<String, Object> parameters = getParameters(context, config);
+                final Map<String, Object> parameters = getParameters(context, config, CONNECTORS_KEYS_MAP);
 
                 final String binding = HTTPConnectorDefinition.SOCKET_BINDING.resolveModelAttribute(context, config).asString();
                 bindings.add(binding);
-                parameters.put(HTTPConnectorDefinition.SOCKET_BINDING.getName(), binding);
                 parameters.put(TransportConstants.HTTP_UPGRADE_ENABLED_PROP_NAME, true);
+                parameters.put(TransportConstants.HTTP_UPGRADE_ENDPOINT_PROP_NAME, HTTPConnectorDefinition.ENDPOINT.resolveModelAttribute(context, config).asString());
+                // uses the parameters to pass the socket binding name that will be read in ActiveMQServerService.start()
+                parameters.put(HTTPConnectorDefinition.SOCKET_BINDING.getName(), binding);
                 connectors.put(connectorName, new TransportConfiguration(NettyConnectorFactory.class.getName(), parameters, connectorName));
             }
         }

--- a/messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties
+++ b/messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties
@@ -174,6 +174,7 @@ connector-service.remove=Operation removing a connector service
 connector-service=TODO
 connector.add.params=A set of key-value pairs understood by the connector factory-class and used to configure it.
 connector.add=Operation adding a connector
+connector.endpoint=The http-acceptor that serves as the endpoint of this http-connector.
 connector.factory-class=Class name of the factory class that can instantiate the connector.
 connector.params=A key-value pair understood by the connector factory-class and used to configure it.
 connector.remove=Operation removing a connector

--- a/messaging-activemq/src/main/resources/schema/wildfly-messaging-activemq_1_0.xsd
+++ b/messaging-activemq/src/main/resources/schema/wildfly-messaging-activemq_1_0.xsd
@@ -260,6 +260,7 @@
                     </xs:sequence>
                     <xs:attribute name="name" type="xs:string" use="required" />
                     <xs:attribute name="socket-binding" type="xs:string" use="required" />
+                    <xs:attribute name="endpoint" type="xs:string" use="required" />
                 </xs:complexType>
             </xs:element>
 

--- a/messaging-activemq/src/main/resources/subsystem-templates/messaging-activemq.xml
+++ b/messaging-activemq/src/main/resources/subsystem-templates/messaging-activemq.xml
@@ -17,14 +17,11 @@
             <?ADDRESS-SETTINGS?>
 
             <http-connector name="http-connector"
-                            socket-binding="http">
-                <param name="http-upgrade-endpoint"
-                       value="http-acceptor"/>
-            </http-connector>
+                            socket-binding="http"
+                            endpoint="http-acceptor" />
             <http-connector name="http-connector-throughput"
-                            socket-binding="http">
-                <param name="http-upgrade-endpoint"
-                       value="http-acceptor-throughput"/>
+                            socket-binding="http"
+                            endpoint="http-acceptor-throughput">
                 <param name="batch-delay"
                        value="50"/>
             </http-connector>

--- a/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_1_0.xml
+++ b/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_1_0.xml
@@ -122,7 +122,8 @@
                          slow-consumer-threshold="${slow.consumer.threshold:456}"/>
 
         <http-connector name="http"
-                        socket-binding="http">
+                        socket-binding="http"
+                        endpoint="http">
             <param name="batch-delay" value="${batch.delay:50}"/>
         </http-connector>
         <remote-connector name="netty"
@@ -157,7 +158,7 @@
                         server-id="${my.server-id:0}"/>
         <acceptor name="myacceptor"
                   factory-class="org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnectorFactory">
-            <param name="batch-delay" value="${batch.delay:50}"/>
+            <param name="batchDelay" value="${batch.delay:50}"/>
         </acceptor>
 
         <broadcast-group name="groupT"

--- a/testsuite/domain/src/test/resources/domain-configs/domain-standard.xml
+++ b/testsuite/domain/src/test/resources/domain-configs/domain-standard.xml
@@ -334,11 +334,8 @@
                         <role name="guest" send="true" consume="true" create-non-durable-queue="true" delete-non-durable-queue="true"/>
                     </security-setting>
                     <address-setting name="#" dead-letter-address="jms.queue.DLQ" expiry-address="jms.queue.ExpiryQueue" max-size-bytes="10485760" page-size-bytes="2097152" message-counter-history-day-limit="10"/>
-                    <http-connector name="http-connector" socket-binding="http">
-                        <param name="http-upgrade-endpoint" value="http-acceptor"/>
-                    </http-connector>
-                    <http-connector name="http-connector-throughput" socket-binding="http">
-                        <param name="http-upgrade-endpoint" value="http-acceptor-throughput"/>
+                    <http-connector name="http-connector" socket-binding="http" endpoint="http-acceptor" />
+                    <http-connector name="http-connector-throughput" socket-binding="http" endpoint="http-acceptor-throughput">
                         <param name="batch-delay" value="50"/>
                     </http-connector>
                     <in-vm-connector name="in-vm" server-id="0"/>
@@ -598,11 +595,8 @@
                         <role name="guest" send="true" consume="true" create-non-durable-queue="true" delete-non-durable-queue="true"/>
                     </security-setting>
                     <address-setting name="#" dead-letter-address="jms.queue.DLQ" expiry-address="jms.queue.ExpiryQueue" max-size-bytes="10485760" page-size-bytes="2097152" message-counter-history-day-limit="10"/>
-                    <http-connector name="http-connector" socket-binding="http">
-                        <param name="http-upgrade-endpoint" value="http-acceptor"/>
-                    </http-connector>
-                    <http-connector name="http-connector-throughput" socket-binding="http">
-                        <param name="http-upgrade-endpoint" value="http-acceptor-throughput"/>
+                    <http-connector name="http-connector" socket-binding="http" endpoint="http-acceptor" />
+                    <http-connector name="http-connector-throughput" socket-binding="http" endpoint="http-acceptor-throughput">
                         <param name="batch-delay" value="50"/>
                     </http-connector>
                     <in-vm-connector name="in-vm" server-id="0"/>

